### DISCUSSION TITLE
LPS displays inconsistent coordinates

### DISCUSSION THREAD

**Discussion by Ericyiyou** [bitcraze/discussions#1616] [Q&A]
Hi, I'm very new to crazyflie and firmware in general, and I was having some trouble with getting a consistent position on the crazyflie 2.0. I first tried using TWS, but the Z coordinate was only reading negative even though the crazyflie was on the ground during the position measurement. I then switched to TDoA 2/3 (which both returned the same crazyflie coordinates), which made the Z coordinate positive, but the Z position was still inaccurate, as it was measuring a meter off the ground. Additionally, the x and y coordinates were still constantly changing. I'm trying to eventually get swarming to work, but I'm not sure how feasible that would be with the current position measurements. Please let me know if there's any more information I should add that would help!

Video of the position using TDoA 3: https://github.com/user-attachments/assets/8b4cc18c-d45c-4304-91d3-4c91bed8107d

**Comment by ToveRumar** [comment]
Hi @Ericyiyou !

When using the LPS system the z position might have some more offset than the x and y position. This is partly due to 
more disturbances closer to the floor. 
You can offset this programmatically to set the first set point a bit higher than the offset and this should allow you to take of the floor, this offset usually corrects itself on higher altitudes. You could mitigate this a bit by trying to have the floor being as empty as possible and not using reflective materials close to the system.

The LPS system has about 10cm accuracy, however this much instability looks more like a setup issue. The accuracy depends a lot on  the placement of the LPS nodes.
If you have not seen this recommendation for setting up the nodes, this could be good information;
https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/#place-the-anchors-in-the-room

I hope this helps in setting up the system and getting a stable position!
